### DISCUSSION TITLE
Expand `≈` arguments like `@test` does

### DISCUSSION THREAD

**Discussion by jariji** [Seelengrab/Supposition.jl#50] [Design & Ideas]
```jl
julia> @test sqrt(2) ≈ identity(1.4)
Test Failed at REPL[177]:1
  Expression: sqrt(2) ≈ identity(1.4)
   Evaluated: 1.4142135623730951 ≈ 1.4
```

It's nice that `@test` shows the `Evaluated: 1.4142135623730951 ≈ 1.4`. I would like that whenever I use `≈` in `@check`.

**Comment by Seelengrab** [comment]
Can you be more specific? What invocation of `@check` should handle this printing in particular? I.e., what's the expected input & expected output?

**Reply by jariji** [reply]
Currently:

```jl
julia> @check function _foo(x=Data.Floats{Float64}())
           sqrt(x^2) ≈ x
       end
Found counterexample
  Context: _foo

  Arguments:
      x::Float64 = 1.3407807929942597e154
```

I want the first result to include

```jl
  Expression: sqrt(x ^ 2) ≈ x
   Evaluated: Inf ≈ 1.3407807929942597e154
```

just like `@test` does:

```jl
julia> let x = 1.3407807929942597e154
           @test sqrt(x^2) ≈ x
       end
Test Failed at tmp.jl:212
  Expression: sqrt(x ^ 2) ≈ x
   Evaluated: Inf ≈ 1.3407807929942597e154
```

**Reply by Seelengrab** [reply]
This is not going to be supported, sorry. There's too much "magic" (for lack of a better word) involved, and there's too many cases where this either fails or ends up printing too much.

Use `event!` to record any additional information that you want to display in the final report.
